manifests/database/postgresql_ssl_rules.pp

```diff
@@ -4,18 +4,24 @@
 define puppetdb::database::postgresql_ssl_rules (
   String $database_name,
   String $database_username,
+  String $postgres_version,
```

Comment:
Can you make the datatype a bit more strict, at least `String[2]`?

Reply:
Oh or we could make it Float? That would make more sense I think?

Reply:
I think that's introducing unrelated changes because technically the module still supports postgresql 9. Another PR to update/fix all of the postgres_version types passing around.

Reply:
It might still support 9.6, but nothing earlier, so that's a String with 3 chars or a valid Float. Afterwards postgresql changed their versioning scheme. Now I haven't tested If 11 is a valid Float in puppet or if it has to be 11.0. (if we drop 9.6 we could use Integer as datatype).

Reply:
My decision to set the datatype as `String` was based on it being (implied) as `String` in the rest of the module. The `$postgres_version` variable is available as a parameter with no explicit type in manifests/init.pp, with the default value taken from manifests/params.pp, where it is set (to `'11'` or `'9.6'`) based on the requested version of PuppetDB; given the quotes, the value we're going to be getting is a `String` of some kind. Since all I had to do to solve the clientcert problem was plumb the value taken from init.pp down to manifests/database/postgresql_ssl_params.pp, I wanted to keep changes minimal, so I kept the implied type of `String`.

Whether `$postgres_version` should be `Float` or `Integer` instead of `String` is a worthwhile question IMO but also, I humbly submit, outside the scope of this change, as there would me numerous other changes needed as a result. Changing that would also break backwards compatibility for end users (me, for example) who had previously set it as a `String` value, would it not?  The tests still pass if I change it to `String[2,3]` (to accommodate e.g. '16' or '9.6'), so please let me know if you'd like me to go ahead and push that change.

Reply:
I think `String[2,3]` is a good way to move forward.